manifests/server.pp

```diff
@@ -383,6 +380,13 @@
     'clusterManager', 'clusterMonitor', 'hostManager', 'root', 'restore',
   ],
 ) inherits mongodb::globals {
+  if $journal != undef {
+    if ($mongodb::globals::manage_package_repo and versioncmp($mongodb::globals::repo_version, '7.0') >= 0)
+    or ($package_ensure =~ /\./ and versioncmp($package_ensure, '7.0.0') >= 0 ) {
+      fail('`journal` parameter is only supported for MongoDB < 7.0')
+    }
+  }
+
```

Comment:
Does this mean that `mongodb::globals::repo_version` needs to be set or the `package_ensure` variable needs to be set to a version for this validation to work? The validation is just skipped when someone manages their own repo, right?

Reply:
> The validation is just skipped when someone manages their own repo

That was my intention. We can't validate something we don't know about.
I added the version constraint in param doc just for this.

But I now see that there is a small bug.
When we bump the default to 7.0 this will start to fail even if `repo_location`is set.
I'll check that repo_location is undef as well.

Reply:
Fixed and enhanced by looking for a version match in provided `repo_location` as well.
Added a few more tests.